viewer/packages/rendering/src/sector/SceneComponentsProvider.ts

```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+export type NodeTypeExistences = { back: boolean; inFront: boolean; ghost: boolean };
+
+export interface SceneComponentsProvider {
+  splitScene(): NodeTypeExistences;
+  restoreScene(): void;
+
+  getCustomScene(): THREE.Scene;
+
+  getCadScene(): THREE.Scene;
+
+  prepareNormalScene(): void;
+  getNormalScene(): THREE.Scene;
+  restoreNormalScene(): void;
+
+  prepareInFrontScene(): void;
+  getInFrontScene(): THREE.Scene;
+  restoreInFrontScene(): void;
+
+  getGhostScene(): THREE.Scene;
```

Comment:
Any way to reduce this into only two functions? With only `splitScene()` and `restoreScene()`?